### DISCUSSION TITLE
PPT

### DISCUSSION THREAD

**Discussion by Leeeunsookl** [k-mcp/My_Class_Planner#57] [Announcements]
인트로 (Introduction)

프로그램 개요 (Program Overview)
주요 기능 (Key Features)
사용자 인터페이스 (User Interface)
동작 원리 (How It Works)
기술 스택 (Technology Stack)
동작 원리 (How It Works)
데모 (Demo)
추가 기능 개발 예정 & 프로젝트 리뷰(참여도 한달 동안으로 잡아서 인사이트에서 추가) 
Q&A (Questions & Answers)

<슬라이드 내용>

인트로 (Introduction)
Title: 자동 시간표 생성 프로그램
Subtitle: 효율적인 시간 관리 도구
Author: [Team Members]

프로그램 개요 (Program Overview)
목적: 과목명과 강의를 입력하면 가능한 조합으로 시간표를 자동 생성
대상 사용자: 학생, 교육기관 관리자 등
효과: 시간 절약, 효율적인 시간 관리

프로젝트 기획
기획 배경 : 시장조사(현재 미국에서 사용하는 대학의 LMS시스템 및 학교 고문 직무 *노션 사진들 내용)
유사경쟁사 (1. 노션 캡쳐본 가볍게 넘김 왜냐하면 대부분 미국 대학교의 LMS 사용하고 있고 UI도 대부분 비슷함. 
2. 에브리타임 (시간표마법사 에브리타임을 전신으로 만듦. 에브리타임의 시작 + 현재 수익등) 국가별 대학 시스템에는 큰 차이가 없기 때문에 니즈는 충분하다는 판단 + 시장 규모 

 
기획 의도
기획 목표

개발
개발 기간 및 프로세스
업무 분장

주요 기능 (Key Features)
과목 및 강의 입력: 과목명과 해당 강의를 저장
자동 시간표 생성: 입력된 강의들을 조합하여 최적의 시간표 생성
사용자 맞춤형 설정: (예시) 개인 시간대 및 우선순위 반영

사용자 인터페이스 (User Interface)
스크린샷 1: 과목 및 강의 입력 화면
스크린샷 2: 자동 생성된 시간표 화면
설명: 직관적인 UI로 쉽게 사용 가능

동작 원리 (How It Works)
Step 1: 과목명 및 강의 정보 입력
Step 2: 알고리즘이 가능한 모든 강의 조합 계산
Step 3: (예시) 사용자 설정에 맞춘 최적 시간표 선택
Diagram: 간단한 알고리즘 흐름도

기술 스택 (Technology Stack)
프로그래밍 언어: Springboot
프레임워크: Spring Framework
데이터베이스: MySQL
웹 기술: HTML, CSS, JavaScript (Thymeleaf)
기타 도구: Maven, Git

데모 (Demo)
영상: 프로그램 데모 비디오 링크
설명: 주요 기능 시연

추후 개발 예정 
일주일로 정리된 시간표 api
exp 하루당 맥시멈 점수를 제한할 계획

리뷰 > 공헌도 사진 사용

Q&A (Questions & Answers)
참여 유도: 질문과 피드백 요청
감사 인사: 참여해주셔서 감사합니다

**Comment by Leeeunsookl** [comment]
참고) 랭킹시스템 왜????????
EXP 100000 채우면 관리자라고 되어있는데 정확히 말하면 저희에게 관리자 요청을 할 수 있습니다. 그만큼 많이 사용해봤기 때문에 부족한 부분을 짚어줄 유저의 눈이 필요하고, 직원으로 함께 업무할 수 있도록 일종의 구인제도라고 보면 될 것 같습니다. 
100000 EXP 를 채우면 CONTACTUS로 연락하도록 게시해놨습니다.
 


리뷰
처음에 방식은 가장 잘하는 것을 배분해서 각자 업무를 맡는 식이었으나,
데이터모델링부터 다같이 진행. 백엔드, 프론트엔드 업무 등을 나눠서 다 같이 맡게 됨. 팀원들이 각자 기능을 확인하기 위해 수시로 어플리케이션을 돌려보다가 다른 팀원이 완료해 놓은 코드에 오류나 부족한 부분을 찾았을 경우 원코드를 작성했던 조원에게 오류사실을 알린 뒤 상황에 맞게 다른 팀원이 고친 후 다같이 재확인하는 경우도 많았음, 유연하게 움직였기 때문에 조원들이 각자 업무에 진행중일 때 집중력을 떨어뜨리지 않을 수 있어서 효율적이었음. 
덕분에 원래 목표였던 my plans 시간표 알고리즘 이외에 커뮤니티, 컨택어스, 세팅, 서치 유저까지 기능을 만들고 프론트 통일까지 완성할 수 있었음. 각자 잘하고 못하는 것이 있었지만 함께 참여하며 약한 부분들은 채워나가고 강한부분은 팀을 이끌어나갔음. 팀프로젝트를 통해서  얻은 고찰은, 협업이라는 것이 단순히 개인의 합이 아니라 1+1이 3이 될 수 있다는 것임.

**Comment by Leeeunsookl** [comment]
미리 세팅 되어 있어야 할 것, 
가입자 본인의 가입되지 않은 이메일 로그인 상태로 켜놓기

Contact Us 로 메일을 보내면 확인 할 수 있는 계정도 로그인상태로 켜놓기

**Reply by tesla1220** [reply]
2단계 인증 필요 없는 이메일 계정 -> 가입 시 발송되는 인증 이메일 확인 목적

**Comment by Jungwoo-Ryu** [comment]
발표할때 ai chat 에 "search 유저는 어떤 기능이야?" 라고 질문하기
